bench/app.py

```diff
@@ -81,16 +81,18 @@ def setup_details(self):
 		if not self.is_repo:
 			self.repo = self.app_name = self.name
 			return
+
+		# fetch meta for repo on mounted disk
+		if os.path.exists(self.mount_path):
+			self.on_disk = True
+			self._setup_details_from_mounted_disk()
+
 		# fetch meta from installed apps
-		if self.bench and os.path.exists(os.path.join(self.bench.name, "apps", self.name)):
+		elif self.bench and os.path.exists(os.path.join(self.bench.name, "apps", self.name)):
 			self.mount_path = os.path.join(self.bench.name, "apps", self.name)
 			self.from_apps = True
 			self._setup_details_from_mounted_disk()
 
-		# fetch meta for repo on mounted disk
-		elif os.path.exists(self.mount_path):
```

Comment:
How does changing the order of conditions fix this? 

Is first condition also true previously? :thinking: 

If so, can't we just set self.on_disk to true in that condition?

Reply:
>Is first condition also true previously?

Yes. The [previous first condition](https://github.com/frappe/bench/blob/develop/bench/app.py#L85-L88) was getting true previously.

>If so, can't we just set self.on_disk to true in that condition?

I tested it and this will work as well. In that case I think [this condition](https://github.com/frappe/bench/blob/develop/bench/app.py#L91-L93) won't be necessary.

---

I have found another bug while testing the above. - If dependent apps are already installed, new app installation fails while running the `get-app` command with `--resolve-deps`.
Eg. If I already have ERPNext installed in the Bench, and I try installing custom_app then the installation fails. Here, custom_app has ERPNext as dependency. Same goes for installing `hrms` with `--resolve-deps`.

I have fixed this new bug as well, testing currently.